web/modules/custom/speech/src/Controller/SinchTicketGeneratorController.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Drupal\speech\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Site\Settings;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Generates Sinch ticket.
+ */
+class SinchTicketGeneratorController extends ControllerBase {
+
+  /**
+   * The application key.
+   *
+   * @var string
+   */
+  private $applicationKey;
+
+  /**
+   * The application secret key.
+   *
+   * @var string
+   */
+  private $applicationSecret;
+
+  /**
+   * The current user.
+   *
+   * @var AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * SinchTicketGeneratorController constructor.
+   *
+   * @param AccountInterface $current_user
+   *   The current user.
+   */
+  public function __construct(AccountInterface $current_user) {
+    $this->applicationKey = Settings::get('sinch-key');
+    $this->applicationSecret = Settings::get('sinch-secret');
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static (
+      $container->get('current_user')
+    );
+  }
+
+  /**
+   * Manages the content on the ticket generator page.
+   *
+   * @return Response
+   */
+  public function content() {
+    // The current time.
+    $created_at = new \DateTime('now');
+
+    // The user ticket.
+    $user_ticket = [
+      'identity' => [
+        'type' => 'username',
+        'endpoint' => $this->currentUser->getAccountName(),
+      ],
+      'expiresIn' => 3600,
+      'applicationKey' => $this->applicationKey,
+      'created' => $created_at->format('c'),
+    ];
+
+    // Json encodes the user ticket.
+    $user_ticket_json = preg_replace('/\s+/', '', json_encode($user_ticket));
```

Comment:
This method doesn't respect SOLID. It creates the user ticket, it encodes the ticket and returns a response.